homeassistant/components/obihai/connectivity.py

```diff
@@ -0,0 +1,66 @@
+"""Support for Obihai Connectivity."""
+from __future__ import annotations
+
+from pyobihai import PyObihai
+
+from .const import DEFAULT_PASSWORD, DEFAULT_USERNAME, LOGGER
+
+
+def get_pyobihai(
+    host: str,
+    username: str,
+    password: str,
+) -> PyObihai:
+    """Retrieve an authenticated PyObihai."""
+    return PyObihai(host, username, password)
+
+
+def validate_auth(
+    host: str,
+    username: str,
+    password: str,
+) -> bool:
+    """Test if the given setting works as expected."""
+    obi = get_pyobihai(host, username, password)
+
+    login = obi.check_account()
+    if not login:
+        LOGGER.debug("Invalid credentials")
+        return False
+
+    return True
+
+
+class ObihaiConnection:
+    """Contains a list of Obihai Sensors."""
+
+    def __init__(
+        self,
+        host: str,
+        username: str = DEFAULT_USERNAME,
+        password: str = DEFAULT_PASSWORD,
+    ) -> None:
+        """Store configuration."""
+        self.sensors: list = []
+        self.host = host
+        self.username = username
+        self.password = password
+        self.serial: list = []
+        self.services: list = []
+        self.line_services: list = []
+        self.call_direction: list = []
+        self.pyobihai: PyObihai = None
+
+    def update(self) -> bool:
+        """Validate connection and retrieve a list of sensors."""
+        self.pyobihai = get_pyobihai(self.host, self.username, self.password)
```

Comment:
Should this not check if pyobihai already exists?
```suggestion
        if not self.pyobihai:
            self.pyobihai = get_pyobihai(self.host, self.username, self.password)
```

And maybe `check_account` can also be guarded?